src/Microsoft.Health.Fhir.Core/Exceptions/EverythingOperationException.cs

```diff
@@ -4,21 +4,29 @@
 // -------------------------------------------------------------------------------------------------
 
 using System.Diagnostics;
+using System.Net;
 using Microsoft.Health.Fhir.Core.Models;
 
 namespace Microsoft.Health.Fhir.Core.Exceptions
 {
     public class EverythingOperationException : FhirException
     {
-        public EverythingOperationException(string message)
+        public EverythingOperationException(string message, HttpStatusCode httpStatusCode, string contentLocationHeaderValue = null)
             : base(message)
         {
             Debug.Assert(!string.IsNullOrEmpty(message), "Exception message should not be empty.");
 
+            ResponseStatusCode = httpStatusCode;
+            ContentLocationHeaderValue = contentLocationHeaderValue;
+
             Issues.Add(new OperationOutcomeIssue(
                 OperationOutcomeConstants.IssueSeverity.Error,
                 OperationOutcomeConstants.IssueType.Invalid,
                 message));
         }
+
+        public HttpStatusCode ResponseStatusCode { get; }
```

Comment:
I don't think we've coupled HttpStatusCode inside Exceptions yet, our codebase currently maps these. I'm fine if we've thought about it, is EverythingOperationException becoming too generic?

Reply:
I noticed that `OperationFailedException` and `FhirTransactionFailedException` have a status code property - however I can definitely make changes if we think `EverythingOperationException` is becoming too generic. Thoughts on the best way to break it down? We could either rely on an existing exception type or create another new exception class (open to either).